.devcontainer/README.md

```diff
@@ -0,0 +1,217 @@
+
+### **5. GitHub Authentication**
+
+Shortly after starting, the Dev Container will prompt you to authenticate with GitHub. While optional, this authentication grants access to NVIDIA's distributed `sccache`, significantly accelerating builds. However, this feature is currently exclusive to NVIDIA employees who are members of the `NVIDIA` or `rapidsai` GitHub organizations.
+
+Follow the prompt, which will have you enter a one-time code at https://github.com/login/device. Follow the instructions on the GitHub web page and after succesfully authenticating, your GitHub account is used to authenticate access to the sccache distributed storage. Succesfully completing this process looks like the following:
+
+![Shows authentication with GitHub to access sccache bucket.](./img/github_auth.png)
+
+This step is entirely optional and will not impact any functionality other than not being able to read or write to the distributed sccache storage. `sccache` will still write a local cache to your filesystem that will still accelerate local rebuilds.
+
+If at any time you want to run this authentication process manually, you can do so by running the `devcontainer-utils-vault-s3-init` script inside the container.
+
+For more information about the sccache configuration and authentication, see the documentation at [`rapidsai/devcontainers`](https://github.com/rapidsai/devcontainers/blob/branch-23.10/USAGE.md#build-caching-with-sccache).
+
+### **6. Working Inside the Devcontainer**
+
+With the container launched, you're now in an environment tailored for CCCL development. Whether coding, running tests, or conducting other tasks, you'll benefit from the tailored tools and settings of this environment.
+
+For more information on building and running tests, see the [contributer guide](../CONTRIBUTING.md).
+
+## Using Docker Images Directly (Manual Approach)
+
+While Visual Studio Code with Dev Containers offers a smooth development experience, there might be developers who lean towards a direct Docker approach. For those, we've simplified the process with our `launch.sh` script.
+
+Before using the script, you might want to explore the available configurations.
+The `.devcontainer` directory and its subdirectories house these configurations, with each subdirectory typically named as `cuda<CTK_VERSION>-<HOST-COMPILER>`.
+
+### **Steps**:
+
+1. **Use the `launch.sh` Script**:
+   To initiate your desired development environment, run the `launch.sh` script with flags specifying your desired CUDA and compiler versions:
+   ```bash
+   ./launch.sh --cuda 12.2 --host gcc12 --docker
+   ```
+   This command pulls the appropriate Docker image, runs it, and provides an interactive shell within the container.
+
+2. **Workspace Setup**:
+   The `launch.sh` script automatically mounts the CCCL repository directory into the container. This ensures you have a mirrored workspace inside the container, resembling your local setup.
```

Comment:
```suggestion
   The `launch.sh` script automatically mounts the CCCL repository directory into the container. It can be accessed in the container through the `/workspace` directory. This ensures you have a mirrored workspace inside the container, resembling your local setup.
```